RubberduckTests/CodeExplorer/CodeExplorerTests.cs

```diff
@@ -2872,48 +1257,440 @@ public void CompareByNodeType_FoldersAreSortedByName()
         }
 
-        #region Helpers
-
-        private RubberduckParserState TestParserState(Mock<IVBE> vbe, IProjectsRepository projectsRepository)
+        protected class MockedCodeExplorer : IDisposable
         {
-            var vbeEvents = MockVbeEvents.CreateMockVbeEvents(vbe);
-            return new RubberduckParserState(vbe.Object, projectsRepository, new DeclarationFinderFactory(), vbeEvents.Object);
-        }
+            private readonly GeneralSettings _generalSettings = new GeneralSettings();
 
-        private Configuration GetDefaultUnitTestConfig()
-        {
-            var unitTestSettings = new UnitTestSettings(BindingMode.LateBinding, AssertMode.StrictAssert, true, true, false);
+            private readonly Mock<IUiDispatcher> _uiDispatcher = new Mock<IUiDispatcher>();
+            private readonly Mock<IConfigProvider<GeneralSettings>> _generalSettingsProvider = new Mock<IConfigProvider<GeneralSettings>>();
+            private readonly Mock<IConfigProvider<WindowSettings>> _windowSettingsProvider = new Mock<IConfigProvider<WindowSettings>>();
+            private readonly Mock<ConfigurationLoader> _configLoader = new Mock<ConfigurationLoader>(null, null, null, null, null, null, null, null);
+            private readonly Mock<IVBEInteraction> _interaction = new Mock<IVBEInteraction>();
+            private readonly Mock<IFolderBrowserFactory> _browserFactory = new Mock<IFolderBrowserFactory>();
 
-            var generalSettings = new GeneralSettings
+            private MockedCodeExplorer()
             {
-                EnableExperimentalFeatures = new List<ExperimentalFeatures>
+                _generalSettingsProvider.Setup(s => s.Create()).Returns(_generalSettings);
+                _windowSettingsProvider.Setup(s => s.Create()).Returns(WindowSettings);
+                _configLoader.Setup(c => c.LoadConfiguration()).Returns(GetDefaultUnitTestConfig());
+
+                SaveDialog = new Mock<ISaveFileDialog>();
+                SaveDialog.Setup(o => o.OverwritePrompt);
+
+                OpenDialog = new Mock<IOpenFileDialog>();
+                OpenDialog.Setup(o => o.AddExtension);
+                OpenDialog.Setup(o => o.AutoUpgradeEnabled);
+                OpenDialog.Setup(o => o.CheckFileExists);
+                OpenDialog.Setup(o => o.Multiselect);
+                OpenDialog.Setup(o => o.ShowHelp);
+                OpenDialog.Setup(o => o.Filter);
+                OpenDialog.Setup(o => o.CheckFileExists);
+
+                FolderBrowser = new Mock<IFolderBrowser>();
+                _browserFactory
+                    .Setup(m => m.CreateFolderBrowser(It.IsAny<string>(), true,
+                        @"C:\Users\Rubberduck\Documents\Subfolder")).Returns(FolderBrowser.Object);
+            }
+
+            public MockedCodeExplorer(string code) : this(ProjectType.HostProject, ComponentType.StandardModule, code) { }
+
```

Comment:
By all rights, this overload should work just fine. It doesn't. This has a handful of failing tests, probably related to the setup of `VbComponent`. If anyone has any insight, feel free to share. :confused: